File: src/console/log/logger/cache/LogMessagesCache.php
<?php

namespace pocketcloud\cloud\console\log\logger\cache;

use pocketcloud\cloud\console\log\CloudLogger;

final class LogMessagesCache {

    public const int MAX_LINES_IN_MEMORY = 100;

    private static array $savedLines = [];

    public static function save(string $line): void {
        self::$savedLines[] = $line;
        if (count(self::$savedLines) > self::MAX_LINES_IN_MEMORY) {
            self::$savedLines = array_slice(self::$savedLines, -self::MAX_LINES_IN_MEMORY);
        }
    }

    public static function clear(): void {
        self::$savedLines = [];
    }

    public static function print(): void {
        foreach (self::$savedLines as $line) {
            CloudLogger::get()->echo($line);
        }
    }
}

File: src/main/java/de/pocketcloud/cloud/console/log/cache/LogMessagesCache.java
package de.pocketcloud.cloud.console.log.cache;

import de.pocketcloud.cloud.console.log.CloudLogger;

import java.util.ArrayDeque;
import java.util.Deque;

public final class LogMessagesCache {

    public static final int MAX_LINES_IN_MEMORY = 100;

    private static final Deque<String> savedLines = new ArrayDeque<>();

    public static synchronized void save(String line) {
        savedLines.addLast(line);
        while (savedLines.size() > MAX_LINES_IN_MEMORY) {
            savedLines.removeFirst();
        }
    }

    public static synchronized void clear() {
        savedLines.clear();
    }

    public static synchronized void print() {
        for (String line : savedLines) {
            CloudLogger.get().echo(line);
        }
    }
}